### ISSUE TITLE
Unexpected EIR values when using single speed air conditioners performance curves

### ISSUE THREAD

**Issue by dmey** [NREL/EnergyPlus#7185]
I am currently testing EneryPlus [CalcDoe2DXCoil](https://github.com/NREL/EnergyPlus/blob/develop/src/EnergyPlus/DXCoils.cc) with [these](https://github.com/NREL/EnergyPlus/blob/develop/datasets/ResidentialACsAndHPsPerfCurves.idf#L33-L123) curves for single speed air conditioners originally generated by [Cutler et al. (2013)](https://www.nrel.gov/docs/fy13osti/56354.pdf) however I am getting negative values of EIR when the indoor wet-bulb temperature (IWB) is approximately equal to and greater than 30 degree Celsius (C). Has anyone else had similar issue with these curves? Below is a plot of EIR function of IWB and outdoor dry-bulb temperature -- note the very low/negative values of EIR for IWB <= 30 C. 

![image](https://user-images.githubusercontent.com/26348635/53130694-e4034f00-3562-11e9-8128-9726f69e5a36.png)

**Comment by jonwinkler** [comment]
I believe this is a classic quadratic extrapolation issue. The return IWB data Dylan used to develop the performance curves ranged from 13.9-22.2 C, which are typical return conditions in U.S. residences. I don't think the curves should be used beyond this range despite the limits far exceeding this range in the EnergyPlus curve objects.

**Comment by dmey** [comment]
I see -- thanks. A couple of questions/points:
- Would this not imply that the AC is always on and able to keep up with the demand to keep the temperature in range 13.9-22.2 C? How realistic is this? How would you handle cases where the AC is turned on when the  IWB > 22.2?

- Would you know where I can find similar well-used/-known curves for single-speed ACs that cover a wider range?  Interestingly, the capacity curves look reasonable -- are these extrapolated at 22.2+ C?
![image](https://user-images.githubusercontent.com/26348635/53133880-c33ff700-356c-11e9-9cab-9cf623e7c7b4.png)

- I think it would be good to add a comment in the `ResidentialACsAndHPsPerfCurves.idf` file as to state the valid ranges and perhaps add checks. I note ` !- Minimum Curve Output` and ` !- Maximum Curve Output` however these will simply set the curve to e.g. 0 (current preset) if it's negative. I only noticed this issue with the curves because I was getting very strange values of COP. I may be alone here but this restriction with the temperature range was unknown to me previously,

**Comment by rraustad** [comment]
Best bet is to acquire manufacturer extended performance data and use that instead (if available). Your right that a simulation will typically extrapolate under certain conditions. There's nothing that can be done to restrict that except to set up the simulation model to stay within the curve boundaries (e.g., use RH control). What is usually recommended is that the min/max x and y be populated with the data set limits, but then when you exceed those limits the curve output does not change from what it is at those limits, so that answer is also wrong. One little known or discussed limitation of empirical models.

**Comment by rraustad** [comment]
One more thing to understand. Looking at the curves above the difference between 20F and 15F is very small compared to the other higher WB temps. This is because as the indoor WB is reduced the coil starts to become dry (I seem to recall 43F as the dry coil limit in AHRI 210/240 but can't be sure right now). When it becomes dry there will no longer be a change in the performance as the indoor WB temp is further reduced. Now in the regression model, as WB temp increases, the impact on capacity or EIR starts increasing more rapidly. It's very hard for a regression to pick up that change without having actual data to use during the regression. If you fit just the lower WB temps (e.g., 15-25) then the coefficients represent that change very well but start loosing accuracy as WB temp increases well past the data limit. Just another reason you need extended data or some method to limit the curve output to reasonable values.

**Comment by dmey** [comment]
Using the curves from the manufacturers make sense but I think that the main issue with these curves included in the EnergyPlus repo is that the minimum and maximum curve output values are not representative of the actual curve/coefficients used for that curve. It looks like these are just arbitrary numbers [0,5] for capacity and EIR function of Temp and [0,2] for capcity and EIR function of flow fraction for all curves in [ResidentialACsAndHPsPerfCurves.idf](https://github.com/NREL/EnergyPlus/blob/develop/datasets/ResidentialACsAndHPsPerfCurves.idf). Is this supposed to be like this? I think that having the correct coefficients in one case but leaving the bounds to incorrect values may lead to unacceptable results.

**Comment by rraustad** [comment]
The coefficients in that file come from the Cutler [report](https://www.nrel.gov/docs/fy13osti/56354.pdf). Appendix A in the report on page 30 has coefficients in degrees C and those coefficients match as does the flow fraction coefficients in the main body of the report. The min/max limits on the curve do not seem to match the limited example data provided in the report. So I suspect that the author of the ResidentialACsAndHPsPerfCurves.idf file arbitrarily entered those limits. Not quite sure how to resolve this without finding the author of that data sets file. Even if the author is found I assume that person won't know the exact curve's data set limits without seeing the data set used to create the performance curves. This might take some time to resolve. For now you should just be aware of the issue and note this somewhere in the repo or report you intend to publish.

**Comment by dmey** [comment]
That's correct -- I have contacted Cutler and waiting to hear back about the minimum and maximum curve output values for HP. For AC, even though these are not explicitly indicated, they seem to be 57-72F EWB and 75-115F ODB but I am awaiting confirmation -- not sure how long this will take. 
Assuming that 57-72F EWB and 75-115F ODB are correct we can calculate the min and max for capacity and EIR function of Temp as follows:

- Capacity

![image](https://user-images.githubusercontent.com/26348635/53572398-1284b980-3b63-11e9-95b4-be1c3b29bb87.png)

- EIR

![image](https://user-images.githubusercontent.com/26348635/53572473-35af6900-3b63-11e9-8a40-1eedb6f3e472.png)

Bu as you said this may take a while to resolve so if you agree, I think that we can start by removing the values from the [ResidentialACsAndHPsPerfCurves.idf](https://github.com/NREL/EnergyPlus/blob/develop/datasets/ResidentialACsAndHPsPerfCurves.idf) file. I have done this in #7197. Let me know if this is OK or needs to be changed in the PR.

**Comment by rraustad** [comment]
Yes, this is correct and I would suggest moving forward. The only fundamental problem I have with using performance data limits to limit curve output results is what happens in the simulation when these limits are exceeded? For example, the AC is turned off at night and has a large load to pull down in the morning, how has the indoor conditions changed (depends on internal gains) and what is the OAT in the morning (cool I suspect)? It's obvious what happens in your range test (15-45 IWB and 20-45 OAT) what problems can arise, but what confidence in the simulation results do we have when we say "if outdoor temp falls below 75F then use the CapFT result at 75". It should be rare when cooling is needed below 75F but it does happen. So how often does IWB or OAT exceed these limits in a simulation? EnergyPlus doesn't care either so your in good (or bad depending on how you look at it) company. Also, how did this affect the results Cutler presented in his paper?

**Comment by dmey** [comment]
When the limits are exceeded the model resets the curve to either min or max value -- the AC is therefore capped to a certain capacity or EIR even though in reality this would not be the case but, at least, this would remove any issues that may arise from extrapolating the values from these curves. I think that the main point of this issue (and the PR) is to make sure that the curves are used as intended and minimise the risk of users to use curves which produce unreasonable results under certain conditions.

**Comment by rraustad** [comment]
I agree. At least you see my point.

**Comment by mjwitte** [comment]
> minimise the risk of users to use curves which produce unreasonable results under certain conditions.

That's exactly why the current limits are in place - to prevent nonsense values (like a negative) or a wildly high value from being used. I disagree with #7197 for that reason. Just add some comments to explain that the limits are arbitrary until there is real basis for new values. 

However, I have always disagreed with clipping a curve to match the range of a dataset (as the curve-fit spreadsheet tool does). Some extrapolation is appropriate and often necessary. If a dataset is too narrow and does not extrapolate well, then it's not useful.

Regardless of what the limits are, it's important to add tracking and reporting of how often and how far the input and output limits are exceeded.

**Comment by dmey** [comment]
@mjwitte the current limits are arbitrary. Why is the max not set to 4, or 6 for example? What's the motivation behind this choice? From the calculations I added above, leaving the max to 5 would mean to extrapolate the curve more than 3 times their valid range (e.g. for capacity from 1.57 to 5). Don't you think it would be more appropriate to have representative defaults and let 'expert' users, or users who have particular needs, change the values themselves?

**Comment by dmey** [comment]
I had a reply from Cutler and it appears that the ranges used to construct the HP curves may be: -15 to 60F for ODB and 65-75F for EDB (see page 31 table 3.6 of [this](https://scholar.colorado.edu/cgi/viewcontent.cgi?article=1278&context=cven_gradetds)) -- the ranges are not stated explicitly therefore it's an assumption as it is for the AC curves.

There are still the function of flow fraction curve (quadratic) where the data is missing to calculate their min and max. Has anyone else have any idea of what ranges to use?

Let me know if you want to calculate the correct ranges and update them in #7197. Also what do we want to do for the curves function of flow fraction curve (quadratic) where I have no solid information to compute the ranges.

**Comment by Myoldmopar** [comment]
I'd like to bring this back up to see if there is any path forward for getting #7197 changed so that it can be merged in. I feel like assuming a minimum value of zero is a low-risk option, and we can continue to discuss the maximum value.

I think there are some differences in opinions, but we can find a good path forward.  

@rraustad @mjwitte @dmey
